### ISSUE TITLE
[4.x] Consider setting "--kube-endpoint PRIVATE_ENDPOINT" in generate_kubeconfig.template.sh

### ISSUE THREAD

**Issue by agregory999** [oracle-terraform-modules/terraform-oci-oke#358]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When the operator is going to gain access to the cluster, it uses the command in generate_kubeconfig.template.sh to do so.  There is an option to use the PRIVATE_ENDPOINT even if the cluster is public.  When you change the cluster access (CP subnet) from public to private the default for what server address comes into kubeconfig changes accordingly.  

However, the operator might be best to access the cluster privately always, since it is in the correct VCN and has the ingress rule to do so.   Even in a public control plane cluster, this can be forced to PRIVATE.  Wondering if it should be in all cases, or perhaps an option.

### New or Affected Resource(s)

generate_kubeconfig.template.sh

### Potential Terraform Configuration

```
[opc@operator ~]$ oci ce cluster create-kubeconfig --cluster-id ocid1.cluster.oc1.xxx.xxx --file $HOME/.kube/config --region us-phoenix-1 --token-version 2.0.0  --kube-endpoint PRIVATE_ENDPOINT
```

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

**Comment by hyder** [comment]
Good idea. kubectl on the operator should always use the private endpoint I think

**Comment by hyder** [comment]
closed in #361
